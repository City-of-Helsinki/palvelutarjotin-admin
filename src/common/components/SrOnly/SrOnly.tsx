import classNames from 'classnames';
import * as React from 'react';

import styles from './srOnly.module.scss';
interface Props {
  as?: 'div' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';
  className?: string;
  children?: React.ReactNode;
}

const SrOnly: React.FC<Props> = ({ as: Tag = 'div', children, className }) => {
  return <Tag className={classNames(styles.srOnly, className)}>{children}</Tag>;
};

export default SrOnly;
