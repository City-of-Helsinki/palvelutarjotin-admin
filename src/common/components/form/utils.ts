/* eslint-disable @typescript-eslint/no-explicit-any */
import { FormikErrors, FormikTouched } from 'formik';
import { TFunction } from 'i18next';
import get from 'lodash/get';

/** Get error text
 * @param {Object} errors
 * @param {Object} touched
 * @param {string} name
 * @param {Function} t
 * @return {string}
 */
export const getErrorText = (
  errors: FormikErrors<any>,
  touched: FormikTouched<any>,
  name: string,
  t: TFunction
) => {
  const error: any = get(errors, name);

  return !!error && get(touched, name)
    ? typeof error === 'string'
      ? (t(error) as string)
      : (t(error.key, error) as unknown as string)
    : '';
};
