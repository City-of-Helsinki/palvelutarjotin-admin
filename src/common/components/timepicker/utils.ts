export type TimeObject = { hours: number; minutes: number };

export function formatTime({ hours, minutes }: TimeObject): string {
  const hour = hours.toString().padStart(2, '0');
  const minute = minutes.toString().padStart(2, '0');
  return `${hour}:${minute}`;
}

export function getTimes(interval = 15): string[] {
  const times = [];

  for (let h = 0; h < 24; h += 1) {
    for (let m = 0; m < 60; m += interval) {
      times.push(formatTime({ hours: h, minutes: m }));
    }
  }

  return times;
}

export function getTimeObjects(interval = 15): TimeObject[] {
  const times: TimeObject[] = [];

  for (let h = 0; h < 24; h += 1) {
    for (let m = 0; m < 60; m += interval) {
      times.push({ hours: h, minutes: m });
    }
  }

  return times;
}
