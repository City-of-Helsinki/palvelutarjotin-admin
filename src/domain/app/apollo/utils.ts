import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
