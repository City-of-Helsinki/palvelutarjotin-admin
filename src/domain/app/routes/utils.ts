import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
