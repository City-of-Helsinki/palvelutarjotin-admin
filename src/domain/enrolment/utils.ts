import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
