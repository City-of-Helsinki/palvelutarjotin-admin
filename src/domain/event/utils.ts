import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
