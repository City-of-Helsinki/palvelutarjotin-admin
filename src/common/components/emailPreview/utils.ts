import { TEMPLATE_TYPE } from '../../../domain/enrolment/types';
import { EmailTemplatesQuery } from '../../../generated/graphql';
import { Language } from '../../../types';

export const getTranslatedTemplate = (
  templates: EmailTemplatesQuery | undefined,
  templateType: TEMPLATE_TYPE,
  locale: Language
) => {
  if (!templates) return null;

  const template = templates.notificationTemplates?.edges.find(
    (t) => t?.node?.type === templateType
  );
  const translatedTemplate = template?.node?.translations.find(
    (i) => i?.languageCode.toLowerCase() === locale
  );

  return translatedTemplate;
};
