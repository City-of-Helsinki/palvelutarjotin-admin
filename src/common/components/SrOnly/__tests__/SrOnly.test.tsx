import * as React from 'react';

import { render, screen } from '../../../../utils/testUtils';
import SrOnly from '../SrOnly';

test('matches snapshot 1', () => {
  const { container } = render(<SrOnly as="h1">Test</SrOnly>);

  const srElement = screen.getByText('Test');
  expect(srElement.tagName).toBe('H1');

  expect(container).toMatchSnapshot();
});

test('matches snapshot 2', () => {
  const { container } = render(<SrOnly as="h2">Test</SrOnly>);

  const srElement = screen.getByText('Test');
  expect(srElement.tagName).toBe('H2');

  expect(container).toMatchSnapshot();
});

test('matches snapshot 3', () => {
  const { container } = render(
    <SrOnly className="testClass" as="span">
      Test
    </SrOnly>
  );

  const srElement = screen.getByText('Test');
  expect(srElement.tagName).toBe('SPAN');

  expect(srElement).toHaveClass('srOnly');
  expect(srElement).toHaveClass('testClass');

  expect(container).toMatchSnapshot();
});
