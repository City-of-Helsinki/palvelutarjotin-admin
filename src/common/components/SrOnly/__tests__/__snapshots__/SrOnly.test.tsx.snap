// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div>
  <h1
    class="srOnly"
  >
    Test
  </h1>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div>
  <h2
    class="srOnly"
  >
    Test
  </h2>
</div>
`;

exports[`matches snapshot 3 1`] = `
<div>
  <span
    class="srOnly testClass"
  >
    Test
  </span>
</div>
`;
